### ISSUE TITLE
Unsure about the local flag

### ISSUE THREAD

**Issue by BrunoBoehm** [stevetweeddale/gatsby-source-git#20]
Hi there, 

the remote repo I clone is quite heavy and I often need to `gatsby clean` in order to make it work. 

Usually when not clearing the cache I get this type of errors, for the record if it helps! 
```bash
ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Field "video" is not defined by type MdxFrontmatterFilterInput.

File: src/pages/articles.js:47:46


 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unknown argument "fields" on field "mdx" of type "Query".

GraphQL request:3:9
2 |   query($slug: String!) {
3 |     mdx(fields: { slug: { eq: $slug } }) {
  |         ^
4 |         frontmatter {

File: src/templates/article.js:38:9


 ERROR #85923  GRAPHQL

There was an error in your GraphQL query:

Cannot query field "desc" on type "MdxFrontmatter".

If you don't expect "desc" to exist on the type "MdxFrontmatter" it is most likely a typo.
```

Anyway as explained, it seems like this error goes away as soon as I `gatsby clean`. My problem is that it takes around 10minutes recloning and building given the size of the repo (and media). For this reason the `local` option looks exactly what I need. I would like to have that remote repo content cloned into a "/content" folder at the root of my gatsby directory. 

Issue: I've added the `local: '/content'` option but when doing my `gatsby build` the repo still gets cloned inside of the cache folder. 

Expected behavior: The cloned repo shoudl be cloned into the `local: "/content"` folder as indicated in the options.

1. Could you be a bit more explicit about how to use that option and have the repo cloned into the relevant local repo as indicated in the options?
2. Can we improve the doc in order to explain how one should go about updating that repo to reflect state of the cloned branch on github?

Thanks!

**Comment by stevetweeddale** [comment]
Hmm, you're right – "explicit path" in the docs should really read "absolute path" (eg it's not relative to the Gatsby root). So adding `local: '/content'` to the options would result in the repo trying to be cloned into `/content` at the root of your drive. That documentation fix would address point 1.

Regarding point 2: 

> Note that using a location outside the cache will prevent you changing the branch via gatsby-config.js. You will need to synchronise the branch of the local checkout yourself.

This just means changing the branch in `gatsby-config.js` won't change the locally checked out branch. You'll need to run `git checkout <new branch>` yourself from within the local clone. I'm guessing that's not clear enough?

**Comment by BrunoBoehm** [comment]
Hey! Thanks for the quick answer. 

Ideally I would need the content from my remote github repo to be cloned at the root of the gatsby project, in a `/content` folder. 

I have indeed tried adding `local: '/content'` or `local: './content'` but when running gatsby develop, the repo is always cloned into `.cache/gatsby-source-git/<repo-name>`.

1. How should I proceed in order for this repo to be cloned outside of the `.cache` folder and into that root `/content` folder.

2. If the repo is cloned into that new /content folder, what will be required to keep it in sync with the upstream repo - erasing it and re-running `gatsby develop` should work ?

Thanks a lot for clarifying!

**Comment by stevetweeddale** [comment]
Ah… apologies, I've only just got around to publishing the version with the `local` setting available (see #18), so my guess is you've got the old version installed from npm, but are looking at the docs on github (which are newer than those on the NPM page). As such, your setting will be ignored. Sorry about that – most folks are probably better than me at publishing new versions to NPM quickly!

> How should I proceed in order for this repo to be cloned outside of the .cache folder and into that root /content folder.

Assuming my above guess is correct, update to version 1.1.0!

> If the repo is cloned into that new /content folder, what will be required to keep it in sync with the upstream repo - erasing it and re-running gatsby develop should work?

Running `gatsby develop` will [run a fetch and hard reset](https://github.com/stevetweeddale/gatsby-source-git/blob/8b9e566a1773dd235f387051b11bbff5d797d67c/gatsby-node.js#L35-L37) on the local clone, wherever it is. It just won't change the branch automatically if you update the `branch` option in `gatsby-config.js` like it normally would.
